flattener/flatten.cc

```diff
@@ -45,9 +45,6 @@ unique_ptr<ast::Expression> extractClassInit(core::Context ctx, unique_ptr<ast::
         it = klass->rhs.erase(it);
     }
 
-    if (inits.empty()) {
-        return nullptr;
```

Comment:
What would you think about keeping this branch around, but instead returning `make_unique<ast::EmptyTree>()`?

Reply:
neat idea! I like it. Let me try that.

Reply:
Hmm.. that way I get this failure:
```
[2019-10-16 01:21:31.517] [fatalFallback] [error] Exception::raise(): core/GlobalState.cc:1011 enforced condition num > 0 has failed: num == 0, name overflow

[2019-10-16 01:21:32.786] [fatalFallback] [error] Backtrace:
sorbet::core::GlobalState::lookupNameUnique(sorbet::core::UniqueNameKind, sorbet::core::NameRef, unsigned short) const (in libcore.so) (GlobalState.cc:1011)
sorbet::core::GlobalState::lookupStaticInitForFile(sorbet::core::Loc) const (in libcore.so) (GlobalState.cc:1608)
sorbet::flatten::FlattenWalk::preTransformClassDef(sorbet::core::Context, std::__1::unique_ptr<sorbet::ast::ClassDef, std::__1::default_delete<sorbet::ast::ClassDef> >) (in libflattener.so) (flatten.cc:82)
```

Reply:
Seems like a bug that was previously being papered over::

https://github.com/sorbet/sorbet/blob/e3c665b1225350af4a255679d56ce718b8e3b906/core/GlobalState.cc#L1608

^ indicates that we're trying to lookupStaticInitForFile on a FileRef that doesn't exist (i.e., has `id() == 0`).

Reply:
cc @DarkDimius is this what you were talking about for sharing static-inits between classes?

Reply:
Oh I see it. It sort of is, but isn't.

Reply:
EmptyTrees have `Loc::none()` hard coded for them so I need to find something else to go here. Maybe empty `InsSeq`?

Reply:
What do we use locs from these trees for?